pyrs/core/summary_generator_stress.py

```diff
@@ -0,0 +1,200 @@
+            column_names = 'vx, vy, vz, d0, d0_error '
+            # directional variables
+            for field_3dir in SummaryGeneratorStress.fields_3dir:
+                for direction in SummaryGeneratorStress.directions:
+                    column_names += field_3dir + '_Dir' + direction + ', '
+                    column_names += field_3dir + '_Dir' + direction + '_error, '
+
+            column_names = column_names[:-2] + '\n'
+            handle.write(column_names)
+            return
+
+        def _write_summary_csv_body(handle):
+
+            def _write_number(number) -> str:
+                if math.isnan(number):
+                    return ', '
+                return str(number) + ', '
+
+            def _write_field_3d(row: int, field: str):
+                entries = ''
+                for direction in SummaryGeneratorStress.directions:
+                    if field == 'Strain':
+                        # TODO add check for strain?
+                        strain = self._get_strain_field(direction)
+                        assert(isinstance(strain, StrainField))
+
+                        strain_value = strain.values[row]
+                        strain_error = strain.errors[row]
+                        entries += _write_number(strain_value) + _write_number(strain_error)
+
+                    elif field == 'Stress':
+                        self._stress.select(direction)
+                        stress_value = self._stress.values[row]
+                        stress_error = self._stress.errors[row]
+                        entries += _write_number(stress_value) + _write_number(stress_error)
+
+                    else:
+                        peak_collection = self._get_peak_collection(direction)
+                        assert(isinstance(peak_collection, PeakCollection))
+
+                        if field == 'd':
+                            d_value = peak_collection.get_dspacing_center()[0][row]
+                            d_error = peak_collection.get_dspacing_center()[1][row]
+                            entries += _write_number(d_value) + _write_number(d_error)
+
+                        elif field == 'FWHM':
+                            fwhm_value = peak_collection.get_effective_params()[0]['FWHM'][row]
+                            fwhm_error = peak_collection.get_effective_params()[1]['FWHM'][row]
+                            entries += _write_number(fwhm_value) + _write_number(fwhm_error)
+
+                        elif field == 'Peak_Height':
+                            height_value = peak_collection.get_effective_params()[0]['Height'][row]
+                            height_error = peak_collection.get_effective_params()[1]['Height'][row]
+                            entries += _write_number(height_value) + _write_number(height_error)
+
+                return entries
+
+            # Function starts here
+            body = ''
+            for row, coordinate in enumerate(self._stress.coordinates):
```

Comment:
can you include a comment summarizing what the two nested `for` loops are going to accomplish?